### PR TITLE
chore: simplify the CODEOWNERS for...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

chore: simplify the CODEOWNERS for che-server; remove people who've left; add replacements; remove global owners from subsection (assuming they already have full rights to the repo)

Change-Id: I524aa893c67eacca46cbac71aee38b3098981965
Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nickboldt** [comment]
Should we also remove  @mshaposhnik in favour of @vinokurig ?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,41 +1,21 @@
...
-
-# assembly
-assembly/** @nickboldt
-assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/** @ibuziuk @nickboldt @themr0c
-
-# core
-core/** @nickboldt
+# che.properties (quasi-documentation)
+# Fionn Kelleher (@osslate) is the replacement for Fabrice, but he doesn't seem to have rights to this repo, so comment this out for now
+# assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/** @osslate
```
Comment (ibuziuk) [inline .github/CODEOWNERS:6]: do we really need to the comment?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,41 +1,21 @@
...
-multiuser/sql-schema/**  @mshaposhnik @nickboldt
+multiuser/**  @mshaposhnik
 
 # wsmaster
-wsmaster/** @nickboldt @vinokurig 
-wsmaster/che-core-api-factory/** @mshaposhnik @nickboldt
-wsmaster/che-core-api-factory-shared/**   @mshaposhnik @nickboldt
-wsmaster/che-core-api-workspace-activity/**   @mshaposhnik @nickboldt
-wsmaster/che-core-api-workspace/**  @mshaposhnik @nickboldt
+wsmaster/** @vinokurig @mshaposhnik
```
Comment (ibuziuk) [inline .github/CODEOWNERS:15]: I do not think Maksim will be still owning wsmaster

Comment (mshaposhnik) [reply .github/CODEOWNERS:15]: me neither..:)

File: .github/CODEOWNERS (hunk 1 above)
Comment (nickboldt) [reply .github/CODEOWNERS:6]: well, if we want docs team to eventually have access to che.properties file again, then this is a useful message to the future. :D

File: .github/CODEOWNERS (hunk 2 above)
Comment (nickboldt) [reply .github/CODEOWNERS:15]: If you use the `suggestion` feature we can skip a step and collaborate faster. 
```suggestion
wsmaster/** @vinokurig 
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,41 +1,21 @@
...
 
 # multiuser che
-multiuser/api/**  @nickboldt
-multiuser/integration-tests/**  @mshaposhnik @nickboldt
-multiuser/keycloak/**  @mshaposhnik @nickboldt
-multiuser/machine-auth/**  @mshaposhnik @nickboldt
-multiuser/permission/**  @nickboldt
-multiuser/personal-account/**  @nickboldt
-multiuser/sql-schema/**  @mshaposhnik @nickboldt
+multiuser/**  @mshaposhnik
```
Comment (nickboldt) [inline .github/CODEOWNERS:12]:
```suggestion
multiuser/**  @vinokurig 
```

**Review by ibuziuk — APPROVED** [review]

**Review by l0rd — APPROVED** [review]

**Pull request merged**